File: mvplibrary/src/main/java/com/junwu/mvplibrary/di/module/AppModule.java
package com.junwu.mvplibrary.di.module;

import android.app.Application;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * ===============================
 * 描    述：AppModule
 * 作    者：pjw
 * 创建日期：2017/6/29 19:43
 * ===============================
 */
@Module
public class AppModule {

    private Application mApplication;

    public AppModule(Application mApplication) {
        this.mApplication = mApplication;
    }

    @Singleton
    @Provides
    public Application providerApplication() {
        return mApplication;
    }

}

File: ktmvplibrary/src/main/java/com/junwu/ktmvplibrary/di/module/AppModule.kt
package com.junwu.ktmvplibrary.di.module

import android.app.Application
import dagger.Module
import dagger.Provides
import okhttp3.OkHttpClient
import java.util.concurrent.TimeUnit
import javax.inject.Singleton

/**
 * ===============================
 * 描    述：
 * 作    者：pjw
 * 创建日期：2017/9/25 15:26
 * ===============================
 */
@Module
class AppModule(application: Application) {

    private val mApplication: Application = application

    @Singleton
    @Provides
    fun providerApplication(): Application {
        return mApplication
    }

}